From stable+bounces-114131-greg=kroah.com@vger.kernel.org Thu Feb  6 17:21:32 2025
From: Koichiro Den <koichiro.den@canonical.com>
Date: Fri,  7 Feb 2025 01:20:54 +0900
Subject: Revert "btrfs: avoid monopolizing a core when activating a swap file"
To: gregkh@linuxfoundation.org, stable@vger.kernel.org
Cc: wqu@suse.com, fdmanana@suse.com, dsterba@suse.com
Message-ID: <20250206162055.1387169-1-koichiro.den@canonical.com>

From: Koichiro Den <koichiro.den@canonical.com>

This reverts commit bb8e287f596b62fac18ed84cc03a9f1752f6b3b8.

The backport for linux-6.1.y, commit bb8e287f596b ("btrfs: avoid
monopolizing a core when activating a swap file"), inserted
cond_resched() in the wrong location.

Revert it now; a subsequent commit will re-backport the original patch.

Fixes: bb8e287f596b ("btrfs: avoid monopolizing a core when activating a swap file") # linux-6.1.y
Signed-off-by: Koichiro Den <koichiro.den@canonical.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---
 fs/btrfs/inode.c |    2 --
 1 file changed, 2 deletions(-)

--- a/fs/btrfs/inode.c
+++ b/fs/btrfs/inode.c
@@ -7387,8 +7387,6 @@ noinline int can_nocow_extent(struct ino
 			ret = -EAGAIN;
 			goto out;
 		}
-
-		cond_resched();
 	}
 
 	if (orig_start)
